### ISSUE TITLE
[@typescript-eslint/ban-ts-comment] Should not have additional properties error

### ISSUE THREAD

**Issue by lffg** [typescript-eslint/typescript-eslint#2233]
**Repro**

```js
{
  rules: {
    // ...
    '@typescript-eslint/ban-ts-comment': [
      1,
      {
        'ts-expect-error': 'allow-with-description',
        minimumDescriptionLength: 2
      }
    ],
    // ...
  }
}
```

```TS
// your repro code case
```

I don't actually know if I am missing something, but I just can't understand this error I'm getting:

```text
Oops! Something went wrong! :(

ESLint: 7.3.0

Error: .eslintrc.js » eslint-config-lffg » ./react.js » ./es.js#overrides[0]:
        Configuration for rule "@typescript-eslint/ban-ts-comment" is invalid:
        Value {"ts-expect-error":"allow-with-description","minimumDescriptionLength":2,"ts-ignore":true,"ts-nocheck":true,"ts-check":false} should NOT have additional properties.

    at validateRuleOptions (/Users/luiz/Dev/luizfelipe.dev/node_modules/eslint/lib/shared/config-validator.js:132:19)
    at /Users/luiz/Dev/luizfelipe.dev/node_modules/eslint/lib/shared/config-validator.js:187:9
    at Array.forEach (<anonymous>)
    at validateRules (/Users/luiz/Dev/luizfelipe.dev/node_modules/eslint/lib/shared/config-validator.js:184:30)
    at validateConfigArray (/Users/luiz/Dev/luizfelipe.dev/node_modules/eslint/lib/shared/config-validator.js:312:9)
    at CascadingConfigArrayFactory._finalizeConfigArray (/Users/luiz/Dev/luizfelipe.dev/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js:464:13)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/Users/luiz/Dev/luizfelipe.dev/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js:275:21)
    at FileEnumerator._iterateFilesRecursive (/Users/luiz/Dev/luizfelipe.dev/node_modules/eslint/lib/cli-engine/file-enumerator.js:431:49)
    at _iterateFilesRecursive.next (<anonymous>)
    at FileEnumerator.iterateFiles (/Users/luiz/Dev/luizfelipe.dev/node_modules/eslint/lib/cli-engine/file-enumerator.js:287:49)
```

It is telling that I'm not suposed to add "additional" properties, but as you can see in the first excerpt of code here, I'm just setting two properties (which are described in the [documentation](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-ts-comment.md)):

> [...] should NOT have additional properties.

<sup>An excerpt from the above error message.</sup>

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.3.0` |
| `@typescript-eslint/parser`        | `3.3.0` |
| `TypeScript`                       | `3.9.x` |
| `ESLint`                           | `7.3.0` |
| `node`                             | `6.13.4` |
| `npm`                              | `12.15.0` |

**Comment by bradzacher** [comment]
I'm unable to repro this.
Likely at the time you were using a version less than 3.1.0, when the `minimumDescriptionLength` option was added.

**Comment by lffg** [comment]
Yes, that was the case. I've just forgot about that possibility. I have a project with Gatsby, which was "overriding" the `@typescript-eslint/plugin` version.

Thank you. :)
